test/integration/LdpHandlerWithAuth.test.ts

```diff
@@ -117,4 +117,24 @@ describe.each(stores)('An LDP handler with auth using %s', (name, { storeUrn, se
     response = await resourceHelper.deleteResource('http://test.com/permanent.txt', true);
     expect(response.statusCode).toBe(401);
   });
+
+  it('can add files but not write to them if append is allowed.', async(): Promise<void> => {
+    // Set acl
+    await aclHelper.setSimpleAcl({ read: true, write: false, append: true }, 'agent');
+
+    // Add a file
+    let response = await resourceHelper.createResource(
+      '../assets/testfile2.txt', 'testfile2.txt', 'text/plain', true,
+    );
+    expect(response.statusCode).toBe(201);
+
+    const id = response._getHeaders().location;
+    response = await resourceHelper.performRequestWithBody(
+      new URL(id),
+      'PUT',
+      { 'content-type': 'text/plain', 'transfer-encoding': 'chunked' },
+      Buffer.from('data'),
+    );
+    expect(response.statusCode).toBe(401);
+  });
```

Comment:
We really need to clear up the test helpers though to make things more readable.
Hard to understand what's going on.

Something like `performPostRequest` or so would help.

Reply:
Yea there are problems with it, problem here is that the helper function is too specific. There is a helper function for doing a PUT, but it also immediately checks the status code which would fail here.

Related to that: we might also want to merge the LdpHandler tests with the Server tests, there's a lot of overlap there.